### PR TITLE
8696 RADx MARS conditional receiver transforms

### PR DESCRIPTION
**Pull request by victor-chaparro**
This PR includes the conditionals needed to support the RADx MARS HL7 v2 format. The RADx format expects Observations to appear in the following order: The COVID-19 result should be the first OBX segment, followed by any other result OBX segments, and then the Age, Sympotmatic, and Date of Onset OBX segments. A follow up ticket to create an Observation based on the patient's age has been created [here](https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/9533).

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:

1. Send a test report using the attached Postman collection
[RADx MARS.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11520229/RADx.MARS.postman_collection.txt)

2. Verify that the request was accepted
3. Open the generated HL7 file in the local sftp folder.
4. Verify the RADx MARS conditionals listed in column H  in the spreadsheet [here](https://docs.google.com/spreadsheets/d/15dKu5zSY23-VVL0oJphILQyHjyDoj_WmTmXUxGXinzQ/edit#gid=0) appear on the HL7 message
5. Verify that OBX segments appear in this order: The COVID-19 result should be the first OBX segment, followed by any other result OBX segments, Sympotmatic, and Date of Onset OBX segments.
6. Run integration tests and make sure they pass 

## Changes
- Added RADx MARS receiver transforms
- Added integration tests for the RADx MARS format

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #8696 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #9531 
- #9532 
- #9533 
- #9623 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Review by mkalish — COMMENTED** [review]
Looks pretty good to me!  Was able to run through the test steps and confirm most of the conditionals.  I think it's likely my own misunderstanding, but the two I wasn't able to see were:
| | | | |
|-|-|-|-|
|2|	Symptomatic assessment value type	|Conditional	|	When OBX-5 (Set 3) is valued: "CWE"|
|3.2|Symptomatic assessment code description	|Conditional|		When OBX-5 (Set 3) is valued: "Whether the patient has symptoms related to condition of interest"|

as discussed on slack we'll huddle on Monday to confirm.

File: prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-base.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+hl7Class: ca.uhn.hl7v2.model.v251.message.ORU_R01
+
+constants:
+  radxMarsPatient: 'Bundle.entry.resource.ofType(Patient)'
+extends: ../ORU_R01-base
+elements:
+  - name: patient-race-text
```
[the remark this answers is not in the document]
Comment (victor-chaparro) [reply prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-base.yml:10]: There's a separate ticket to implement the missing values.

File: prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-base.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    condition: '%resource.extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.exists()'
+    value: ['"https://www.hl7.org/fhir/v2/0005/"']
+
+  - name: patient-race-coding-system
+    condition: '%resource.extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.exists()'
+    value:
+      - '"HL70005"'
```
[the remark this answers is not in the document]
Comment (victor-chaparro) [reply prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-base.yml]: Updated to keep syntax consistent.

File: prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-observation-result.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+constants:
+  hl7ObservationPath: '/PATIENT_RESULT/ORDER_OBSERVATION(%{orderIndex})/OBSERVATION(%{resultIndex})'
+  symptomaticAoe: '%service.supportingInfo.resolve().where(code.coding.code = %symptomaticAoeLoinc)'
+elements:
+
+  # RADx MARS requirement:
+  # Only set the illness onset date if symptomatic AOE answer is Yes
```
Comment (mkalish) [inline prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-observation-result.yml:7]: Just calling out that the spec specifically says to only set the date if it matches the format `{YYYYMMDDHHMMSS[+/-ZZZZ]}` and example bundle has it as `{YYYYMMDD}`

**Review by mkalish — APPROVED** [review]
LGTM!  Before merging, could we create a follow up ticket the captures all the questions that we discussed in the PR and in our huddle yesterday?

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**